generate/generate.go

```diff
@@ -747,7 +747,50 @@ func (s *service) generateHelperFuncs(a *API) {
 				pn("")
 			}
 		}
-
+		if found := hasIDAndProjectidParamField(a.Params); found {
+			ln := strings.TrimPrefix(a.Name, "list")
+			// Generate the function signature
+			pn("// This is a courtesy helper function, which in some cases may not work as expected!")
+			p("func (s *%s) Get%sByIDAndProject(id string, projectid string,", s.name, parseSingular(ln))
+			for _, ap := range a.Params {
+				if ap.Required && s.parseParamName(ap.Name) != "id" {
+					p("%s %s, ", s.parseParamName(ap.Name), mapType(ap.Type))
+				}
+			}
+			pn(") (*%s, int, error) {", parseSingular(ln))
+			// Generate the function body
+			pn("	p := &List%sParams{}", ln)
+			pn("	p.p = make(map[string]interface{})")
+			pn("")
+			pn("	p.p[\"id\"] = id")
+			pn("	p.p[\"projectid\"] = projectid")
+			for _, ap := range a.Params {
+				if ap.Required {
+					pn("	p.p[\"%s\"] = %s", s.parseParamName(ap.Name), s.parseParamName(ap.Name))
+				}
+			}
+			pn("")
+			pn("	l, err := s.List%s(p)", ln)
+			pn("	if err != nil {")
+			pn("		if strings.Contains(err.Error(), fmt.Sprintf(")
+			pn("			\"Invalid parameter id value=%%s due to incorrect long value format, \"+")
+			pn("				\"or entity does not exist\", id)) {")
+			pn("			return nil, 0, fmt.Errorf(\"No match found for %%s: %%+v\", id, l)")
+			pn("		}")
+			pn("		return nil, -1, err")
+			pn("	}")
+			pn("")
+			pn("	if l.Count == 0 {")
+			pn("	  return nil, l.Count, fmt.Errorf(\"No match found for %%s: %%+v\", id, l)")
+			pn("	}")
+			pn("")
+			pn("	if l.Count == 1 {")
+			pn("	  return l.%s[0], l.Count, nil", ln)
+			pn("	}")
+			pn("  return nil, l.Count, fmt.Errorf(\"There is more then one result for %s UUID: %%s!\", id)", parseSingular(ln))
```

Comment:
Maybe add the project ID in here as well: `fmt.Errorf(\"There is more then one result for %s UUID in project %%s: %%s!\", projectid, id)`

Should be impossible, but if it ever happends it's a little clearer.